File: debug.c
#include <stdio.h>

int fib(int n) {
    if (n <= 1)
        return 1; 
    return fib(n - 1) + fib(n - 2);
}

int main() {
    int n;
    printf("Enter the value of n: ");
    scanf("%d", n); 

    if (n < 0) {
        printf("Error: Negative input is not allowed.\n"); 
    }

    int dp[10]; 
    for (int i = 0; i < 10; i++) {
        dp[i] = -1; 
    }

    int result = fib(n);
    printf("Fibonacci number is %d\n", result);
    return 0;
}

File: debug.java
import java.util.Scanner;

public class Fibonacci {
    static int fib(int n, int[] dp) {
        if (n <= 1)
            return 1; 
        if (dp[n] != 0)
            return dp[n];
        dp[n] = fib(n - 1, dp) + fib(n - 2, dp);
        return dp[n];
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n: ");
        int n = sc.nextInt();

        if (n < 0) {
            System.out.println("Error: Negative input is not allowed."); 
        }

        int[] dp = new int[10]; 
        System.out.println("Fibonacci calculation done."); 
        fib(n, dp); 
        sc.close();
    }
}
